### ISSUE TITLE
Cart, Checkout, and Order Confirmation: replace mock data with real data in the editor

### ISSUE THREAD

**Issue by pmcpinto** [woothemes/woocommerce#42092]
**Description:** in the Editor, we have a mix of real and mock data, which can create a confusing experience for the merchant. We should introduce real data whenever possible, to ensure merchants accurately represent what will be displayed in the front-end.

**Acceptance criteria:**
- [ ] Order summary: this section should include real product data. Let's display the first product created in the store. If the store has a product with cross-sells, we should use it and display the cross-sells in the Cart Cross-Sells Products area. We only show mock data if the store has 0 products.
- [ ] Contact information: we should show the empty field or the email address of the first customer.
- [ ] Shipping and billing address: we should display the empty field (with the country assigned to the store country) or the information of the first customer.
- [ ] Express checkout: this is a complex area, and this topic has a separate issue: https://github.com/woocommerce/woocommerce/issues/42353. It should considered out of scope.
- [ ] Shipping options: the shipping options should reflect the methods defined in settings, including the block-based local pickup.
- [ ] Payment options: no changes are needed, as we already display the real options.
- [ ] These changes should also be applied to the cart and order confirmation

**Comment by pmcpinto** [comment]
@elizaan36 re: customer data. Do you think we should display data from a real customer of the empty fields? For instance, Shopify displays empty fields, the only exception is the country/region. It defaults to the store country.

**Comment by elizaan36** [comment]
@pmcpinto All the acceptance criteria you have listed above makes sense. I think populating the form with data from real customers will make it more confusing.

For the merchant view I'd lean towards matching what the shopper will see upon landing on the checkout for the first time, with the default country set to store country as you mentioned.

**Comment by pmcpinto** [comment]
@elizaan36 thanks for sharing your feedback.

> For the merchant view I'd lean towards matching what the shopper will see upon landing on the checkout for the first time, with the default country set to store country as you mentioned.

Good point. I'll update the AC.

How should we address this in the Order Confirmation view?

Here's the current placeholder

<img width="1884" alt="Captura de ecrã 2023-11-23, às 10 49 33" src="https://github.com/woocommerce/woocommerce-blocks/assets/3380168/6972c921-668f-4af4-ac22-b81f67913d9b">

**Comment by elizaan36** [comment]
I think the placeholder content is good, the only thing that's distracting is the "Test address 1/Test address 2". Could we replace that with a real address format like "123 Main Street"? The customer name and maybe phone number placeholders would also make it more realistic.

<img width="829" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/a3b6efba-721d-4fd5-85ba-bfb0116b4b99">


Also, I see the "Additional information" section at the bottom of the page. Would this always show the skeleton outline or would the section be populated with what's being pulled in from the 3PD extension? I only think this section should appear if there's actual preview content, otherwise it doesn't seem useful to the merchant. Unless I'm missing something here, WDYT?

<img width="1643" alt="Screenshot 2023-11-29 at 12 31 48 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/cc760916-5551-44ab-86fe-b97e2218093e">

**Comment by pmcpinto** [comment]
> Could we replace that with a real address format like "123 Main Street"? The customer name and maybe phone number placeholders would also make it more realistic.

Sounds good to me.

> Would this always show the skeleton outline or would the section be populated with what's being pulled in from the 3PD extension?

I'm not sure if previewing what's added by extensions is possible. @mikejolley can you confirm?

>  I only think this section should appear if there's actual preview content, otherwise it doesn't seem useful to the merchant. Unless I'm missing something here, WDYT?

I think we should keep it, even if it's not possible to preview the content, as it allows users to reorder the 3rd party content.

**Comment by mikejolley** [comment]
> I'm not sure if previewing what's added by extensions is possible. @mikejolley can you confirm?

Importing 3PD content could lead to a messy or broken view because that could be anything, and it may require certain context. If the skeleton is not suitable for this view we could rework it or include some other kind of placeholder (perhaps on focus rather than always visible?)

**Comment by elizaan36** [comment]
Yes, it makes sense to keep the section preview. The skeleton creates some suspense though 😆 Since it indicates that something will soon appear it's probably not the most suitable indication here. I know this pattern is used in the payment methods block but since it's on a per field basis the user understands what the block is used for. I still think we should move towards matching as close as possible to the view that the new shopper will see.

In this case with the order confirmation I think some small placeholder indication of the content will be helpful. I'm playing around with a section outline that integrates cleanly into the grid but uses a dotted border and lighter gray text to indicate that this is third party content and it can't be previewed.

<img width="1041" alt="Screenshot 2023-12-08 at 11 05 59 AM" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/aa23f1de-800a-4555-9e5b-3fd56986e4fe">

Edit: The text color and border color is consistent with other sections. I think the dotted border makes sense to differentiate this content because as far as I understand, in the shopper experience the 3PD field will likely not have the consistent border to the other sections.

**Comment by elizaan36** [comment]
I'm sort of brainstorming live in this issue, but the exploration above made me think of a way to surface custom fields clearly in the editor. The plugin icon could be used to indicate any information coming from a third party.

<img width="1440" alt="Surface custom fields in the order confirmation - Additional information" src="https://github.com/woocommerce/woocommerce-blocks/assets/5121465/4301dc79-52cd-4e58-9784-63f6b66eea47">

**Comment by pmcpinto** [comment]
I like this option as it feels more aligned with the rest of the placeholder data. IMO the skeleton outline is too heavy.
